realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -162,6 +164,33 @@ public String getClassName() {
     public abstract RealmObjectSchema addRealmListField(String fieldName, RealmObjectSchema objectSchema);
 
     /**
+     * Adds a new field that references a {@link RealmList} with primitive values supported by Realm.
+     * See {@link RealmObject} for the list of supported types. Nullability of elements are either defined
+     * by using the correct boxed or primitive type e.g., {@code Integer.class} instead of {@code int.class} or
+     * {@link #setRequired(String, boolean)}.
+     * <p>
+     * Example:
+     * <pre>
+     * {@code
+     * // Defines the list of Strings as being non null.
+     * RealmObjectSchema schema = schema.create("Person")
+     *     .addRealmListField("children", String.class)
+     *     .setRequired("children", true)
+     * }
+     * </pre>
+     * If the list contains references to other Realm classes, then use
+     * {@link #addRealmListField(String, RealmObjectSchema)} instead.
+     *
+     * @param fieldName name of the field to add.
+     * @param elementType simple type of elements in the array.
+     * @return the updated schema.
+     * @throws IllegalArgumentException if the field name is illegal, a field with that name already exists or
+     * the element type isn't supported.
+     * @throws UnsupportedOperationException if this {@link RealmObjectSchema} is immutable.
+     */
+    public abstract RealmObjectSchema addRealmListField(String fieldName, Class<?> elementType);
```

Comment:
I think naming `elementType` to be `primitiveType` would make this method easier to understand, that way you don't need to read the javadoc to know that this is for array of primitives only.